drivers/led/pca9633.c

```diff
@@ -204,16 +205,24 @@ static int pca9633_led_init(const struct device *dev)
 	return 0;
 }
 
-static struct pca9633_data pca9633_led_data;
-
 static const struct led_driver_api pca9633_led_api = {
 	.blink = pca9633_led_blink,
 	.set_brightness = pca9633_led_set_brightness,
 	.on = pca9633_led_on,
 	.off = pca9633_led_off,
 };
 
-DEVICE_DT_INST_DEFINE(0, &pca9633_led_init, NULL,
-		    &pca9633_led_data,
-		    NULL, POST_KERNEL, CONFIG_LED_INIT_PRIORITY,
-		    &pca9633_led_api);
+#define PCA9633_DEVICE(id) \
+	static const struct i2c_dt_spec pca9633_##id##_cfg = {	\
+		.bus = DEVICE_DT_GET(DT_INST_BUS(id)),		\
+		.addr     = DT_INST_REG_ADDR(id),			\
```

Comment:
adrr member shouldn't be needed anymore in case of using the i2c_dt_spec struct.